### PR TITLE
Add friendly exceptions + support Link and Meta tag objects + adopt to new yiisoft/view

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

Adopt to https://github.com/yiisoft/view/pull/154

Successfully tested with demo application.

### CODE REVIEW

**Review by devanych — APPROVED** [review]

File: src/Exception/InvalidLinkTagException.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public function getName(): string
+    {
+        return 'Invalid link tag configuration in injection';
+    }
+
+    public function getSolution(): ?string
+    {
+        return 'Got link tag:' . "\n" . var_export($this->tag, true);
```
Comment (samdark) [inline src/Exception/InvalidLinkTagException.php:45]: The idea here is to provide steps to fix the problem, not more details. There should be an example on how to register a tag.

File: src/Exception/InvalidLinkTagKeyException.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function getName(): string
+    {
+        return 'Invalid link tag key in injection';
+    }
+
+    public function getSolution(): ?string
+    {
+        return 'Got link tag configuration:' . "\n" . var_export($this->tag, true);
```
Comment (samdark) [inline src/Exception/InvalidLinkTagKeyException.php:40]: The idea here is to provide steps to fix the problem, not more details. There should be an example on how to register a tag.

File: src/Exception/InvalidLinkTagPositionException.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function getName(): string
+    {
+        return 'Invalid link tag position in injection ';
+    }
+
+    public function getSolution(): ?string
+    {
+        return 'Got link tag configuration:' . "\n" . var_export($this->tag, true);
```
Comment (samdark) [inline src/Exception/InvalidLinkTagPositionException.php:40]: The idea here is to provide steps to fix the problem, not more details. There should be an example on how to register a tag.

File: src/Exception/InvalidMetaTagException.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public function getName(): string
+    {
+        return 'Invalid meta tag configuration in injection';
+    }
+
+    public function getSolution(): ?string
+    {
+        return 'Got meta tag:' . "\n" . var_export($this->tag, true);
```
Comment (samdark) [inline src/Exception/InvalidMetaTagException.php:45]: The idea here is to provide steps to fix the problem, not more details. There should be an example on how to register a tag.

File: src/Exception/InvalidMetaTagKeyException.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function getName(): string
+    {
+        return 'Invalid meta tag key in injection';
+    }
+
+    public function getSolution(): ?string
+    {
+        return 'Got meta tag configuration:' . "\n" . var_export($this->tag, true);
```
Comment (samdark) [inline src/Exception/InvalidMetaTagKeyException.php:40]: The idea here is to provide steps to fix the problem, not more details. There should be an example on how to register a tag.

File: src/LinkTagsInjectionInterface.php [hunk 6]
```diff
@@ -4,27 +4,54 @@
...
-     *         'rel' => 'icon',
-     *         'type' => 'image/png',
-     *         'href' => '/myicon.png',
+     *         Html::link('/myicon.png', [
+     *             'rel' => 'icon',
+     *             'type' => 'image/png',
+     *         ]),
+     *     ],
+     *     [
+     *         '__key' => 'themeCss',
```
Comment (samdark) [inline src/LinkTagsInjectionInterface.php:38]: Can we use array key as key instead of `__key`?

File: src/ViewRenderer.php [hunk 7]
```diff
@@ -207,19 +234,72 @@ private function renderProxy(
         );
     }
 
+    /**
+     * @psalm-param MetaTagsConfig $tags
+     */
     private function injectMetaTags(array $tags): void
     {
+        /** @var mixed $tag */
         foreach ($tags as $tag) {
```
Comment (samdark) [inline src/ViewRenderer.php:243]:
```suggestion
        foreach ($tags as $key => $tag) {
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Exception/InvalidLinkTagException.php (hunk 1 above)
Comment (vjik) [reply src/Exception/InvalidLinkTagException.php]: Done.

File: src/Exception/InvalidLinkTagKeyException.php (hunk 2 above)
Comment (vjik) [reply src/Exception/InvalidLinkTagKeyException.php]: Done.

File: src/Exception/InvalidLinkTagPositionException.php (hunk 3 above)
Comment (vjik) [reply src/Exception/InvalidLinkTagPositionException.php]: Done.

File: src/LinkTagsInjectionInterface.php (hunk 6 above)
Comment (vjik) [reply src/LinkTagsInjectionInterface.php]: Done.

File: src/ViewRenderer.php (hunk 7 above)
Comment (vjik) [reply src/ViewRenderer.php]: Done.

**Review by samdark — APPROVED** [review]

**Pull request merged**